gaussian_example/transform_base-1.0.yaml

```diff
@@ -0,0 +1,135 @@
+%YAML 1.1
+---
+$schema: http://stsci.edu/schemas/yaml-schema/draft-01
+id: asdf://asdf-format.org/extensions/transform/schemas/transform_base-1.0
+description: >
+  Common validations for transforms.
+
+definitions:
+  parameter_value:
+    description: >
+      Parameter value.  May be a single number, or in the case of a
+      transform set, a 1D ndarray of numbers.
+    anyOf:
+      - type: number
+      - tag: asdf://asdf-format.org/extensions/core/tags/ndarray-*
+        ndim: 1
+
+  parameter:
+    description: >
+      Object that encapsulates properties of a transform parameter.
+    properties:
+      value:
+        description: >
+          Parameter value.
+        $ref: '#/definitions/parameter_value'
+      unit:
+        description: >
+          Parameter unit.
+        default: >
+          Parameter has no unit.
```

Comment:
What is the advantage of separating `value` and `unit`? Why not keep the `quantity` ?

Reply:
A few (not particularly strong) reasons:
- The `unit` seems like more of a property of the parameter and not just the value, since it also applies to the bounds
- (unit, unitless) x (value, ndarray, YAML array) is a tidy way to express all the combinations vs many different value types
- I find the quantity schema difficult to work with.  For example, I can't add that constraint on the array dimensions without doing something like this:

```
dimension_limited_quantity:
  allOf:
    - tag: asdf://asdf-format.org/extensions/core/tags/quantity-*
    - properties:
        value:
          ndim: 1
```
and I can't require a quantity scalar without doing this:
```
quantity_scalar:
  allOf:
    - tag: asdf://asdf-format.org/extensions/core/tags/quantity-*
    - properties:
        value:
          type: number
```